transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
Well, a binary search (by which I mean commenting out big chunks of stuff and seeing if it still breaks) has narrowed things down to [this line](https://github.com/relayrides/pushy/blob/v0.7.3/pushy/src/main/java/com/relayrides/pushy/apns/ApnsClient.java#L452):

```java
pipeline.addLast(sslContext.newHandler(channel.alloc()));
```

It doesn't seem like actually adding the handler is the problem; it's something that happens later. Now that I know where to look, I'll try to reduce this to a minimal test case. I wanted to share the intermediate results in the meantime, though, in case that's enough information to point somebody else in the right direction.